### PR TITLE
fix: bump the min maxUnavailable to 1

### PR DESCRIPTION
**Pull request by ryanzhang-oss**
### Description of your changes

Bump the min maxUnavailable to 1 to avoid rolling out stuck.

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: apis/placement/v1/clusterresourceplacement_types.go [hunk 1]
```diff
@@ -500,7 +500,7 @@ type RollingUpdateConfig struct {
 	// Absolute number is calculated from percentage by rounding up.
 	// We consider a resource unavailable when we either remove it from a cluster or in-place
 	// upgrade the resources content on the same cluster.
-	// This can not be 0 if MaxSurge is 0.
+	// The minimum of MaxUnavailable is 1 to avoid rolling out stuck during in-place resource update.
```
Comment (circy9) [inline apis/placement/v1/clusterresourceplacement_types.go:503]: // The desired number equals to the `NumberOfClusters` field when the placement type is `PickN`.
	// The desired number equals to the number of clusters scheduler selected when the placement type is `PickAll`.
	
What is the desired number? Why does it equal to max possible number?

File: apis/placement/v1/clusterresourceplacement_types.go [hunk 2]
```diff
@@ -500,7 +500,7 @@ type RollingUpdateConfig struct {
...
 	// Defaults to 25%.
 	// +kubebuilder:default="25%"
 	// +kubebuilder:validation:XIntOrString
```
Comment (circy9) [inline apis/placement/v1/clusterresourceplacement_types.go:506]: Does this work?

+kubebuilder:validation:ExclusiveMinimum:true
+kubebuilder:validation:Minimum:0

File: apis/placement/v1/clusterresourceplacement_types.go [hunk 3]
```diff
@@ -513,7 +513,7 @@ type RollingUpdateConfig struct {
 	// The desired number equals to the number of clusters scheduler selected when the placement type is `PickAll`.
```
Comment (circy9) [inline apis/placement/v1/clusterresourceplacement_types.go:513]: // The desired number equals to the `NumberOfClusters` field when the placement type is `PickN`.

What is the desired number?

File: apis/placement/v1/clusterresourceplacement_types.go (hunk 1 above)
Comment (ryanzhang-oss) [reply apis/placement/v1/clusterresourceplacement_types.go:503]: it is used to calculate the real number when the maxUnavailable is a percent like 25%

File: apis/placement/v1/clusterresourceplacement_types.go (hunk 2 above)
Comment (ryanzhang-oss) [reply apis/placement/v1/clusterresourceplacement_types.go:506]: I dobut it as this is not an int

**Review by circy9 — APPROVED** [review]

**Pull request merged**